### PR TITLE
fix value escaping in codeql-env.sh

### PR DESCRIPTION
**Pull request by hmakholm**
CodeQL itself complained (correctly) that the escaping here would fail if the value contains backslashes.

Switch to single-quoting, which is less tricky to escape.

Hmmm, how confident are we that there are not consumers that try to parse `codeql-env.sh` themselves and will break if they don't find double quotes? They _should_ be using the JSON output instead, but still ...

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/master/README.md) has been updated if necessary.

### CODE REVIEW

File: src/runner.ts [hunk 1]
```diff
@@ -249,7 +249,7 @@ program
         const shEnvFileContents = Object.entries(tracerConfig.env)
           // Some vars contain ${LIB} that we do not want to be expanded when executing this script
           .map(
-            ([key, value]) => `export ${key}="${value.replace(/\$/g, "\\$")}"`
+            ([key, value]) => "export " + key + "='" + value.replace(/'/g, "'\"'\"'") + "'"
```
Comment (aeisenberg) [inline src/runner.ts:252]:
```suggestion
            ([key, value]) => `export ${key}='${value.replace(/'/g, "'\"'\"'")}'`
```

**Review by aeisenberg — APPROVED** [review]
Yes, it's safer to use single quotes instead of escaping the `$`.

**Comment by alexet** [comment]
I know there are consumers converting the windows output to sh (as they use bash on windows) but I don't know about users converting the sh output.

File: src/runner.ts [hunk 2]
```diff
@@ -249,7 +249,7 @@ program
         const shEnvFileContents = Object.entries(tracerConfig.env)
           // Some vars contain ${LIB} that we do not want to be expanded when executing this script
           .map(
-            ([key, value]) => `export ${key}="${value.replace(/\$/g, "\\$")}"`
+            ([key, value]) => `export ${key}='${value.replace(/'/g, "'\"'\"'")}'`
```
Comment (aeisenberg) [inline src/runner.ts:252]:
```suggestion
            ([key, value]) =>
              `export ${key}='${value.replace(/'/g, "'\"'\"'")}'`
```

Comment (hmakholm) [reply src/runner.ts:252]: Will that work? The linter apparently said it wants a newline _between_ the backticks; you're adding one _in front of_ them.

Comment (aeisenberg) [reply src/runner.ts]: You can ignore the open/close `. It is just being used to delineate the start and end of the insertion string. 

```
Insert `⏎·············`
```

It makes sense if you see this in an IDE (well, vscode at least), but less sense if you see it as a cli message.

**Comment by aeisenberg** [comment]
Arrrrgh....also need to update the compile files. Let me just push a change up.

**Comment by hmakholm** [comment]
Do we need a change note warning users of the different quoting?

**Comment by aeisenberg** [comment]
Based on how we are doing things now, the answer is _no_. And as we discussed, this might change, but I don't think we should be holding this PR up until we make any process changes.

**Comment by aeisenberg** [comment]
@hmakholm, merged this for you because you mentioned earlier that you kept having merge conflicts if you waited too long.

**Pull request merged**